python/torch_mlir_e2e_test/test_suite/basic.py

```diff
@@ -1748,6 +1748,54 @@ def IndexTensorMultiInputOneDim_basic(module, tu: TestUtils):
 # ==============================================================================
 
 
+class IndexTensorMultiInputContiguousOneDimDynamic(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.float32, True),
+        ([-1, 1], torch.int64, True),
+        ([-1], torch.int64, True),
+    ])
+    def forward(self, x, index1, index2):
+        return torch.ops.aten.index(x, (None, index1, index2,))
+
+
+@register_test_case(module_factory=lambda: IndexTensorMultiInputContiguousOneDimDynamic())
+def IndexTensorMultiInputContiguousOneDimDynamic_basic(module, tu: TestUtils):
+    module.forward(tu.rand(5, 4, 3), torch.randint(4, (6, 1)), torch.randint(3, (3,)))
+
+
+# ==============================================================================
+
+
+class IndexTensorMultiInputNonContiguousOneDimDynamic(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.float32, True),
+        ([-1, 1], torch.int64, True),
+        ([-1], torch.int64, True),
+    ])
+    def forward(self, x, index1, index2):
+        return torch.ops.aten.index(x, (index1, None, index2,))
+
+
+@register_test_case(module_factory=lambda: IndexTensorMultiInputNonContiguousOneDimDynamic())
+def IndexTensorMultiInputNonContiguousOneDimDynamic_basic(module, tu: TestUtils):
+    module.forward(tu.rand(5, 4, 3), torch.randint(4, (6, 1)), torch.randint(3, (3,)))
+
+
+# ==============================================================================
+
+
```

Comment:
Can you add a few of the following cases as tests:
1. index1 and index2 are rank 2 with both dims dynamic, and the dim sizes are different
2. index1, index2, and index3 are all rank 1, and index1 is statically sized at [2] (anything except 1) while index2 and index3 are dynamically sized at [2]
3. Same as one of the above test cases, but swap the order of index1 and index2 when performing the index.Tensor op